src/metatensor/vesin.h

```diff
@@ -0,0 +1,135 @@
+/*INDENT-OFF*/
+#ifndef VESIN_H
+#define VESIN_H
+
+#include <stddef.h>
```

Comment:
@Luthaf do you really need a C compatible header here? If possible I would remote the C case and replace these with includes from the C++ std library (`cstddef` and `cstdint`)

Reply:
This needs to be a C compatible header since the library only exposes a C API.

As discussed above (in the "Unresolved questions/issues" section), this code is included as a library and not developed specially for plumed. Ideally I'd like to exclude it from the code checks in plumed, is this possible? I suggested three potential solutions in the initial message, is there one you prefer?

If nothing else works I can change the code here, but any future update to the library (for bug fixes or better performance, …) will require re-doing the same set of changes.

Reply:
Thanks for pointing to this, I forgot this note in the original pull request.

Notice that you can exclude the codecheck adding `metatensor/` similarly to other external packages  [here](https://github.com/plumed/plumed2/blob/abe78d1d0c2a99875a66486a0d451cad1179a7f5/src/maketools/codecheck#L79)

However, we cannot accept the `vesin` namespace as is, because it adds symbols that are not in the `PLMD` namespace. This could create clashes if e.g. you combine PLUMED with another code using another version of the `vesin` library. If you look at other "external" tools (e.g. molfile, blas, lapack, xdfile, etc) they all provide symbols hidden in the `PLMD` namespace.

Given this, I would suggest:
- writing a script that fixes namespace when importing the vesin library. You might want to get inspiration from `src/*/import.sh` scripts.
- if possible, write such script so that all the issues with codecheck are fixed. if not, we can disable codecheck as explained above
- run the `./header` script on `metatensor` and check if it breaks something

This would be very close to what we do with other external packages.

Reply:
I had another idea that should handle the symbol clash as well: basically move the code to `external/vesin-single-build.cpp` and then add a file `neighbors.cpp` which looks like

```
namespace PLMD {
namespace metatensor {

#include "./external/vesin.h"
#include "./external/vesin-single-build.cpp"

}}
```

I can then add the `metatensor/external` folder to the excluded folder list, and this seems to pass all checks.

Reply:
This would be the only source code file located in a subdirectory so I would avoid it if possible

Reply:
ok, I finally had time to come back to this. The code is now using an `import.sh` file to import code from vesin